flash/core/integrations/icevision/wrappers.py

```diff
@@ -36,7 +39,7 @@ def _effdet_validation_step(validation_step, batch, batch_idx):
 
 def wrap_icevision_adapter(adapter):
     if not isinstance(adapter.log, partial):
-        adapter.log = partial(_log_with_prog_bar_override, adapter.log)
+        adapter.log = partial(_log_with_name_and_prog_bar_override, adapter.log, adapter)
```

Comment:
How about we just pass the stage, since `_log_with_name_and_prog_bar_override` doesn't use `adapter` except for stage?

```suggestion
        adapter.log = partial(_log_with_name_and_prog_bar_override, adapter.log, adapter.trainer.state.stage)
```

Reply:
That won't work as the trainer attribute won't necesarily have been set yet. We also need to reference the stage which changes throughout training